packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -141,6 +142,18 @@ class RefreshIndicator extends StatefulWidget {
   /// its actual displacement may significantly exceed this value.
   final double displacement;
 
+  /// The distance of [RefreshProgressIndicator] from the top or bottom edge of
+  /// its parent.
+  ///
+  /// Depending whether the indicator is showing on the top or bottom, the value
+  /// of this variable controls how far from the target edge will the progress
+  /// indicator start to appear. This may come in handy when, for example, the
+  /// UI contains a top [Widget] which covers the edge where the progress
+  /// indicator would otherwise appear.
+  ///
+  /// By default, the edge distance is set to 0.
+  final double edgeDistance;
```

Comment:
I am still not sure we need another parameter here. I am not that familiar with the design of the`RefreshIndicator`, but it would seem to me these are describing the same functionality. My preference would be to just fix `displacement` to do what your `edgeDistance` is doing. That is move the displacement from the container padding to the `Positioned` parameter, so that it starts off displaced.

If you do really think there is a need for both parameters, can you give better descriptions to each that show when you would use one over the other? Also, I agree that `edgeOffset` would be a better name if we do need both.

@hansmuller, do you see a need for both, or would it make sense to just fix the existing `displacement` to include the displacement from the beginning of the drag?

Reply:
`displacement ` is the distance from the edge(appear) to the last position(disappear).

Reply:
They serve two different things, I don't see how they are the same. Just to add, the UI I detailed above is not possible to achieve with only `displacement` -- one needs both properties.

I will rename to `edgeOffset`.